### ISSUE TITLE
Scaled pods not detected by telegraf in Kubernetes cluster 

### ISSUE THREAD

**Issue by Pranav-Balakumar** [influxdata/telegraf#10148]
## Relevant telegraf.conf:

```
[[inputs.prometheus]]
  metric_version = 2
  ## Scrape Kubernetes pods for the following prometheus annotations:
  ## - prometheus.io/scrape: Enable scraping for this pod
  ## - prometheus.io/scheme: If the metrics endpoint is secured then you will need to
  ##     set this to `https` & most likely set the tls config.
  ## - prometheus.io/path: If the metrics path is not /metrics, define it with this annotation.
  ## - prometheus.io/port: If port is not 9102 use this annotation
  monitor_kubernetes_pods = true
```

## System info:

Telegraf `1.20.4`
Telegraf executable run as a process in a Kubernetes pod in the cluster

## Steps to reproduce:

- Deploy a pod that exposes prometheus metrics at <pod_ip>:<port>/<metrics path> with the annotations in the comment for monitor_kubernetes_pods above.
- Run telegraf with the prometheus input plugin and the settings monitor_kubernetes_pods = true.
- The metrics from the pod will be seen in telegraf.
- Let the telegraf pod run for long duration eg. 24h
- Scale the above pod that produces metrics ( increase replicas )

## Expected behavior:

Telegraf will discover the new scaled pods scrape the metrics exposed by those pods.

## Actual behavior:

New scaled pods aren't discovered/registered , no prometheus metrics are scraped from them.

## Additional info:

This is different from the issue #9600 as its related to new pods, and this issue is related to the scaled pods.

## Workaround:

The scaled pods are discovered if the telegraf pod is restarted.

**Comment by MyaLongmire** [comment]
@Pranav-Balakumar will you please test  #10932?

**Comment by Pranav-Balakumar** [comment]
I have tested it and the fix is working fine

**Comment by powersj** [comment]
Fixed with #10932
